packages/dapp/components/redeem/lib/use-prices.ts

```diff
@@ -1,28 +1,32 @@
-import useDeployedContracts from "@/components/lib/hooks/contracts/use-deployed-contracts";
-import useManagerManaged from "@/components/lib/hooks/contracts/use-manager-managed";
+import { getIMetaPoolContract } from "@/components/utils/contracts";
+import useProtocolContracts from "@/components/lib/hooks/contracts/use-protocol-contracts";
+import useWeb3 from "@/components/lib/hooks/use-web-3";
 import { BigNumber, utils } from "ethers";
 import { useEffect, useState } from "react";
 
 const usePrices = (): [BigNumber | null, BigNumber | null, () => Promise<void>] => {
-  const deployedContracts = useDeployedContracts();
-  const managedContracts = useManagerManaged();
+  const protocolContracts = useProtocolContracts();
+  const { provider } = useWeb3();
 
   const [twapPrice, setTwapPrice] = useState<BigNumber | null>(null);
   const [spotPrice, setSpotPrice] = useState<BigNumber | null>(null);
 
   async function refreshPrices() {
-    if (managedContracts && deployedContracts) {
-      const dollarTokenAddress = await deployedContracts.manager.dollarTokenAddress();
-      const newTwapPrice = await managedContracts.dollarTwapOracle.consult(dollarTokenAddress);
-      const newSpotPrice = await managedContracts.dollarMetapool["get_dy(int128,int128,uint256)"](0, 1, utils.parseEther("1"));
+    if (protocolContracts && provider) {
+      const dollarTokenAddress = protocolContracts.managerFacet && await protocolContracts.managerFacet.dollarTokenAddress();
```

Comment:
same with this, this 
```protocolContracts.managerFacet` && await protocolContracts.managerFacet.dollarTokenAddress();``` 

the && doesn't look architecturaly feasible within a const, please improve it

Reply:
no, i'm referring to the ```protocolContracts.managerFacet && await protocolContracts.managerFacet.dollarTokenAddress();```

const is ok

Reply:
> no, i'm referring to the `protocolContracts.managerFacet && await protocolContracts.managerFacet.dollarTokenAddress();`
> 
> const is ok

Yeah.. I thought that there is no problem with that architecture.. please correct me